esmvalcore/_diagnostics.py

```diff
@@ -0,0 +1,111 @@
+"""Diagnostics and tags management."""
```

Comment:
It's fine to move this code to a separate file, but it would be good to keep all configuration code together in one place. Could you move this file and the original `_config.py` and `_logging.py` to a new directory called `_config`?

Reply:
This was surprisingly tricky. I was wondering if `config-user.yml` and `config-developer.yml` should move along, but I thought it would be better to leave them where they are. Let me know what you think 👍